yarn-project/p2p/src/services/reqresp/protocols/block_txs/block_txs_handler.ts

```diff
@@ -0,0 +1,76 @@
+import { TxArray, type TxWithHash } from '@aztec/stdlib/tx';
+
+import type { PeerId } from '@libp2p/interface';
+
+import type { AttestationPool } from '../../../../mem_pools/attestation_pool/attestation_pool.js';
+import type { TxPool } from '../../../../mem_pools/index.js';
+import type { ReqRespSubProtocolHandler } from '../../interface.js';
+import { ReqRespStatus, ReqRespStatusError } from '../../status.js';
+import { BitVector } from './bitvector.js';
+import { BlockTxsRequest, BlockTxsResponse } from './block_txs_reqresp.js';
+
+/**
+ * Handler for block txs requests
+ * @param attestationPool - the attestation pool to check for block proposals
+ * @param mempools - the mempools containing the tx pool
+ * @returns the BlockTxs request handler
+ */
+export function reqRespBlockTxsHandler(
+  attestationPool: AttestationPool | undefined,
+  txPool: TxPool,
+): ReqRespSubProtocolHandler {
+  /**
+   * Handler for block txs requests
+   * @param msg - the block txs request message
+   * @returns the block txs response message
+   * @throws if msg is not a valid block txs request
+   */
+  return async (_peerId: PeerId, msg: Buffer) => {
+    let request: BlockTxsRequest;
+    try {
+      request = BlockTxsRequest.fromBuffer(msg);
+    } catch (err: any) {
+      throw new ReqRespStatusError(ReqRespStatus.BADLY_FORMED_REQUEST, { cause: err });
+    }
+
+    const blockProposal = await attestationPool?.getBlockProposal(request.blockHash.toString());
+
+    if (!blockProposal) {
+      throw new ReqRespStatusError(ReqRespStatus.NOT_FOUND);
+    }
+
+    if (blockProposal.blockNumber !== request.blockNumber) {
+      throw new ReqRespStatusError(ReqRespStatus.FAILURE, {
+        cause: new Error(
+          `Requested block proposal number: ${request.blockNumber} does not match the proposal number: ${blockProposal.blockNumber}`,
+        ),
+      });
+    }
```

Comment:
I know we should have caught this during the design phase, but maybe we could just remove the block number from the request? After all, block hash is good enough.